.github/workflows/main_wasm.yml

```diff
@@ -0,0 +1,86 @@
+name: main_wasm
```

Comment:
There is actually very little overlap other than the repository sync step. Nothing from the other builds can be reused as it goes through emscription which sets up a a custom environment and separate build area. So it's not that there are additional steps which can be added on top of a the existing C++ build as is the case of adding python bindings.

Maybe a `build_wasm` flag could be added to a single matrix target, but unless there is an cleaner way to do it, each step would require the a condition on it, with again I think only the sync step in common. I'm not sure if you can group steps together which might make it cleaner ?

Finally as for tracking matrix changes, it doesn't really matter which platform it's built on so just one was chosen. The final result should be deployable on different platforms AFAIK. 

I think if we don't want 2 files, build_wasm could be merged in as another job so there are not multiple workflows.
This would centralize things but there is a clear separation of a wasm job and non-wasm one.